auth/api/iam/api.go

```diff
@@ -297,10 +299,50 @@ func (r Wrapper) HandleAuthorizeRequest(ctx context.Context, request HandleAutho
 	// Workaround: deepmap codegen doesn't support dynamic query parameters.
 	//             See https://github.com/deepmap/oapi-codegen/issues/1129
 	httpRequest := ctx.Value(httpRequestContextKey).(*http.Request)
-	params := make(map[string]string)
+	params := make(map[string]interface{})
 	for key, value := range httpRequest.URL.Query() {
-		params[key] = value[0]
+		if len(value) == 1 {
+			params[key] = value[0]
+		} else {
+			params[key] = value
+		}
 	}
+
+	// if the request param is present, JAR (RFC9101, JWT Authorization Request) is used
+	// we parse the request and check the client_id param
+	var signerKid string
+	if rawToken, ok := params[oauth.RequestParam].(string); ok {
+		// Parse and validate the JWT
```

Comment:
function becomes quite long and does too much (request parsing, switching to the right handler)